Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -2254,15 +2254,20 @@ RefPtr<API::Navigation> WebPageProxy::goToBackForwardItem(WebBackForwardListItem
             protectedBackForwardList()->goToItem(item);
     }
 
-    Ref navigation = m_navigationState->createBackForwardNavigation(process().coreProcessIdentifier(), item, m_backForwardList->protectedCurrentItem(), frameLoadType);
+    Ref process = m_process;
+    if (item.url().isNull()) {
```

Comment:
I don't understand under what conditions this would be hit.  Could we add an ASSERT(preferences().siteIsolationEnabled()) inside that it is for site isolation and catch if it is hit without site isolation on?  Also probably a comment indicating what a null item URL means.